lib/index.js

```diff
@@ -85,47 +132,78 @@ const readmeInspector = {
    *
    * @method check
    * @memberOf module:readme-inspector
-   * @returns Promise
-   * @fulfil {ReadmeInfo} - A ReadmeInfo object with the `scoreData` property.
-   * @reject {ReadmeInfo} - A ReadmeInfo object with a defined `error` property.
+   *
+   * @param {object} params - The owner/org, repo, and (optional)
+   *  branch to check.
+   * @param {string} params.owner - The account or organization to which
+   *  the source code belongs.
+   * @param {string} params.repo - The name of the repository in which
+   *  to look for a README.
+   * @param {string} [params.ref={Default branch}] - The name of the
+   *  commit/branch/tag.
+   *  Default: the repository’s default branch (usually master).
+   * @returns {Promise<ReadmeInfo>}
+   * @fulfil {ReadmeInfo} - A {ReadmeInfo} object with the `appraisal` property.
+   * @reject {ReadmeInfo} - A {ReadmeInfo} object with a defined `error` property.
    */
 
-  async check (owner, repo, ref) {
-    const info = await readmeInspector.getReadmeInfo(owner, repo, ref)
+  async check (params = {
+  }) {
+    const info = await readmeInspector.getInfo(params)
     if (info.isPresent) {
-      info.scoreData = await ReadmeScore.for(`${owner}/${repo}`)
+      info.appraisal = await readmeInspector.getAppraisal(`${params.owner}/${params.repo}`)
     }
     return info
   },
 
+  /**
+   * An API proxy for the readme-score-api, which provides quality
+   * measures for README files with the {@link appraisalData AppraisalData}
+   * interface.
+   *
+   * @method getAppraisal
+   * @memberOf module:readme-inspector
+   * @param {string} url - The URL or slug of a repository.
+   *
+   * @returns {Promise<AppraisalData>}
+   * @see https://github.com/clayallsopp/readme-score-api/#readme
+   */
+
+  async getAppraisal (url) {
+    const readmeAppraisal = new ReadmeAppraisal()
+    const appraisal = await readmeAppraisal.for(url)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "appraisal". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1488)

Reply:
# <kbd>**status: won't fix**</kbd> 

ESLint: [**Disallows unnecessary return await (no-return-await)** ![link-external][octicon-link-external]][no-return-await-url]

> ![quote][octicon-quote] Inside an async function, return await is useless. Since the return value of an async function is always wrapped in Promise.resolve, return await doesn’t actually do anything except add extra time before the overarching Promise resolves or rejects. This pattern is almost certainly due to programmer ignorance of the return semantics of async functions.
>
> _no-return-await - Rules._ (2018). _ESLint - Pluggable JavaScript linter._ Retrieved 21 April 2018, from <https://eslint.org/docs/rules/no-return-await>

<!-- ⛔️ Link references ⛔️  -->

[no-return-await-url]: https://eslint.org/docs/rules/no-return-await
[octicon-link-external]: https://cdnjs.cloudflare.com/ajax/libs/octicons/4.4.0/svg/link-external.svg
[octicon-quote]: https://cdnjs.cloudflare.com/ajax/libs/octicons/4.4.0/svg/quote.svg